src/components/sensor-graph.tsx

```diff
@@ -136,16 +137,48 @@ export default class SensorGraph extends React.Component<SensorGraphProps, Senso
         }
         if (nextProps.xEnd !== xEnd || nextProps.xStart !== xStart) {
             stateChanges.xMin = nextProps.xStart;
-            stateChanges.xMax = nextProps.xEnd;
+            stateChanges.xMax = this.isSingleReadBarGraph() ? 7 : nextProps.xEnd;
         }
         if (Object.keys(stateChanges).length > 0) {
             this.setState(stateChanges);
         }
     }
 
+    isSingleReadBarGraph() {
+      const { singleReads, displayType } = this.props;
+      return singleReads && displayType === "bar";
+    }
+
+    processData() {
```

Comment:
We're not sure about this function. The data does need to be reformatted for single-read bar graphs, but how we're doing that here seems a little hacky. Is there are better approach?